probes/https-outbound-probe.js

```diff
@@ -0,0 +1,210 @@
+        // If no callback has been used then end the metrics after returning from the method instead
+        if (aspect.findCallbackArg(methodArgs) == undefined) {
+          // Need to get request method and URL again
+          var options = methodArgs[0];
+          var requestMethod = 'GET';
+          var urlRequested = '';
+          var headers = '';
+          if (typeof options === 'object') {
+            urlRequested = formatURL(options);
+            if (options.method) {
+              requestMethod = options.method;
+            }
+            if (options.headers) {
+              headers = options.headers;
+            }
+          } else if (typeof options === 'string') {
+            urlRequested = options;
+            var parsedOptions = url.parse(options);
+            if (parsedOptions.method) {
+              requestMethod = parsedOptions.method;
+            }
+            if (parsedOptions.headers) {
+              headers = parsedOptions.headers;
+            }
+          }
+
+          // End metrics (no response available so pass empty object)
+          that.metricsProbeEnd(probeData, requestMethod, urlRequested, {}, headers);
+          that.requestProbeEnd(probeData, requestMethod, urlRequested, {}, headers);
+        }
+        return rc;
+      }
+    );
+  }
+  return target;
+};
+
+// Get a URL as a string from the options object passed to https.get or https.request
+// See https://nodejs.org/api/http.html#http_http_request_options_callback
+function formatURL(httpsOptions) {
+  var url;
+  if (httpsOptions.protocol) {
+    url = httpsOptions.protocol;
+  } else {
+    url = 'https:';
+  }
+  url += '//';
+  if (httpsOptions.auth) {
+    url += httpsOptions.auth + '@';
+  }
+  if (httpsOptions.host) {
+    url += httpsOptions.host;
+  } else if (httpsOptions.hostname) {
+    url += httpsOptions.host;
+  } else {
+    url += 'localhost';
+  }
+  if (httpsOptions.port) {
+    url += ':' + httpsOptions.port;
+  }
+  if (httpsOptions.path) {
+    url += httpsOptions.path;
+  } else {
+    url += '/';
+  }
+  return url;
+}
+
+/*
+ * Lightweight metrics probe for HTTPS requests
+ *
+ * These provide:
+ *   time:            time event started
+ *   method:          HTTPS method, eg. GET, POST, etc
+ *   url:             The url requested
+ *   requestHeaders:  The HTTPS headers for the request
+ *   duration:        The time for the request to respond
+ *   contentType:     HTTPS content-type
+ *   statusCode:      HTTPS status code
+ */
+HttpsOutboundProbe.prototype.metricsEnd = function(probeData, method, url, res, headers) {
+  if (probeData && probeData.timer) {
+    probeData.timer.stop();
+    am.emit('https-outbound', {
+      time: probeData.timer.startTimeMillis,
+      method: method,
+      url: url,
+      duration: probeData.timer.timeDelta,
+      statusCode: res.statusCode,
+      contentType: res.headers ? res.headers['content-type'] : 'undefined',
```

Comment:
maybe should be `: undefined`. right now, if there is no res.headers, contentType is a string `'undefined'`, if there are res.headers, but no 'content-type', then its `undefined` (not a string), seems inconsistent.